python/cudf/cudf/core/dataframe.py

```diff
@@ -5542,7 +5545,11 @@ def to_arrow(self, preserve_index=None):
             else:
                 if isinstance(index, cudf.RangeIndex):
                     index = index._as_int_index()
-                    index.name = "__index_level_0__"
+                    index.name = (
+                        index.name
+                        if index.name is not None
+                        else "__index_level_0__"
+                    )
```

Comment:
With a dataframe:

```
df = cudf.DataFrame({"a": [4, 5, 6]}, index=cudf.RangeIndex(3, name="a"))
```

Can you test that `df.to_arrow(preserve_index=True)` correctly round-trips and produces an `IntIndex` with `name == "a"`?

Reply:
Yup that trips things up:

```
NameError: duplicated column name a
```

Sounds like just writing the index name without modification (which is what the dispatch functions were doing before) is going to pretty error-prone - trying this out in pandas, it seems like an option we have available is to give the index a distinct `field_name` in the schema metadata:

```python
import pandas as pd
import pyarrow as pa

df = pd.DataFrame({"a": [4, 5, 6]}, index=pd.RangeIndex(3, name="a"))

pd_table = pa.Table.from_pandas(df, preserve_index=True)
pd_table.schema.metadata[b'pandas']
```

```
b'{"index_columns": ["__index_level_0__"], ..., "columns": [{"name": "a", "field_name": "a", "pandas_type": "int64", "numpy_type": "int64", "metadata": null}, {"name": "a", "field_name": "__index_level_0__", "pandas_type": "int64", "numpy_type": "int64", "metadata": null}], ...'
```

Can look a bit into the construction of the schema with `pa.pandas_compat.construct_metadata` to see if this is something we could control at that point

Reply:
@charlesbluca - Just pushed a PR to your fork in case it helps: https://github.com/charlesbluca/cudf/pull/1